server/frontend.go

```diff
@@ -144,9 +144,35 @@ var controllers = map[string]*pb.NVMeController{}
 
 func (s *server) CreateNVMeController(ctx context.Context, in *pb.CreateNVMeControllerRequest) (*pb.NVMeController, error) {
 	log.Printf("Received from client: %v", in.Controller)
+	subsys, ok := subsystems[in.Controller.Spec.SubsystemId.Value]
+	if !ok {
+		err := fmt.Errorf("unable to find subsystem %s", in.Controller.Spec.SubsystemId.Value)
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	params := NvmfSubsystemAddListenerParams{
+		Nqn: subsys.Spec.Nqn,
+		ListenAddress: struct {
```

Comment:
Is this best practice with go to copy/paste this struct everywhere it's used?

Reply:
I was reading exactly about this today here https://stackoverflow.com/questions/24809235/how-to-initialize-a-nested-struct

you can either like this (dup) or to split and define 2 different structs....

I can't do 2 different structs now, since we auto-generate the code from json...

will keep looking for a better way...